core/src/Cabana_ExecutionPolicy.hpp

```diff
@@ -13,53 +13,219 @@
 #define CABANA_EXECUTIONPOLICY_HPP
 
 #include <Cabana_Macros.hpp>
+#include <impl/Cabana_Index.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <type_traits>
 
 namespace Cabana
 {
-namespace Experimental
-{
 //---------------------------------------------------------------------------//
 /*!
   \class RangePolicy
-  \brief Execution policy over a range of indices.
+  \brief Execution policy over a range of 1d indices.
+
+  Gives a linear range of indices for executing a functor with concurrency
+  over all indices.
 */
-template<int N, class ExecutionSpace>
-class RangePolicy
+template<class ExecutionSpace>
+class RangePolicy : public Kokkos::RangePolicy<ExecutionSpace>
 {
   public:
 
-    static constexpr int array_size = N;
+    using base_type = Kokkos::RangePolicy<ExecutionSpace>;
 
     using execution_space = ExecutionSpace;
 
-    // Range constructor.
+    /*!
+      \brief Range constructor.
+      \param begin The begininning of the 1D range.
+      \param begin The ending of the 1D range.
+    */
     RangePolicy( const std::size_t begin, const std::size_t end )
-        : _begin( begin )
-        , _end( end )
+        : base_type(begin,end)
```

Comment:
Any reason why you wouldn't just do
```C++
using Kokkos::RangePolicy::RangePolicy;
```
Are they constructors from the base you don't want to make visible?

Reply:
I think the begin/end constructor is the only one I have that mirrors a base constructor. The single argument version may also be in `Kokkos::RangePolicy` although the container one is not.

Reply:
> The single argument version may also be in `Kokkos::RangePolicy`

It is not.

One thing I don't like about it is it goes against the behavior one would expect with C++ - that is, default arguments are for trailing parameters.  And it does not save you many keystrokes `RangePolicy(n)` vs `RangPolicy(0, n)`,